projects/core/src/site-context/facade/currency.service.ts

```diff
@@ -1,50 +1,61 @@
 import { Injectable } from '@angular/core';
 import { Store, select } from '@ngrx/store';
 import { Observable } from 'rxjs';
-import { StateWithSiteContext, CurrencyEntities } from '../store/state';
-import {
-  LoadCurrencies,
-  SetActiveCurrency
-} from '../store/actions/currencies.action';
-import {
-  getAllCurrencies,
-  getActiveCurrency
-} from '../store/selectors/currencies.selectors';
+
 import { OccConfig } from '../../occ/config/occ-config';
+import * as fromStore from '../store/index';
 
+/**
+ * Facade that provides easy access to curreny state, actions and selectors.
+ */
 @Injectable()
 export class CurrencyService {
-  readonly currencies$: Observable<CurrencyEntities> = this.store.pipe(
-    select(getAllCurrencies)
+  /**
+   * Represents all the currencies supported by the current store.
+   */
+  currencies$: Observable<fromStore.CurrencyEntities> = this.store.pipe(
+    select(fromStore.getAllCurrencies)
   );
 
-  readonly activeCurrency$: Observable<string> = this.store.pipe(
-    select(getActiveCurrency)
+  /**
+   * Represents the isocode of the active currency.
+   */
+  activeCurrency$: Observable<string> = this.store.pipe(
```

Comment:
It might be better to keep `readonly` mark on those properties, or switch to proposed alternative with private properties + accessors.

Reply:
Most likely we're going to move to the alternative in order to have a consistent API.